### ISSUE TITLE
Installation breaks when using opt setup flag (pqkmeans, oapackage cannot be installed)

### ISSUE THREAD

**Issue by karanprime** [mala-project/mala#454]
I wanted to test how clustering and sampling according to cluster sizes affects the training. However, the package required for that, pqkmeans, cannot be installed from pip, nor by building it from scratch.

Reproduction:
In a fresh environment, I tried installing mala with the optional flag `pip install -e .[opt]`. 

This results in an error `RuntimeError: numpy cannot be imported. numpy must be installed prior to installing OApackage`

For the clustering part, I only need pqkmeans. So I tried to install it separately using `pip install pqkmeans`.
This results in an error for a dependency of pqkmeans called lshash3.
`ERROR: Failed building wheel for lshash3`.
lshash3 hasn't been updated on pypi since 2017.

I then tried building pqkmeans using instructions from their [github page](https://github.com/DwangoMediaVillage/pqkmeans) but that also throws errors. 

OApackage pypi page: https://pypi.org/project/OApackage/
pkqmeans pypi page: https://pypi.org/project/pqkmeans/
lshash3 pypi page: https://pypi.org/project/lshash3/

tl:dr; Installation for OApackage and pqkmeans is broken.

**Comment by karanprime** [comment]
After the hackathon, we discussed the following todos:
- Fix OAPackage dependency
- Remove clustering code from the main code
I will open a PR for this.
